src/desktop/apps/artsy-v2/server.tsx

```diff
@@ -12,6 +13,22 @@ import { searchMiddleware } from "./apps/search/searchMiddleware"
 
 export const app = express()
 
+const isAllowedRoute = route => route !== "/" && route !== "*"
+
+const topLevelMetaRoute = getAppRoutes()[0]
+const allRoutes = flatten(
+  topLevelMetaRoute.children.map(child => {
+    // Only supports one level of nesting per app.
```

Comment:
@mzikherman - can you elaborate a bit on this and how it relates to everything? Like, only one level in our new apps _period_ or just as a validator?

Reply:
Oh yea, this needs to traverse a possibly deeply nested object to construct all the routes that our global router layer knows about.

Those are constructed with children, such as the artist page and it's tabs. So let's say you had a subtab _within_ a tab, that route wouldn't automatically get picked up here and added to the allow-list.

By children, I mean: https://github.com/artsy/force/blob/22040f8e88ba86bb5485eb394b5bec0c4f1c7758/src/v2/Apps/Artist/routes.tsx#L129 kind of thing. Theoretically each of these children could be their own app (with routes, etc.)

Reply:
This is the current output for `allRoutes`:

```
[
  '/artist/:artistID/works-for-sale',
  '/artist/:artistID/auction-results',
  '/artist/:artistID/consign',
  '/artist/:artistID/cv',
  '/artist/:artistID/articles',
  '/artist/:artistID/shows',
  '/artist/:artistID',
  '/artwork/:artworkID/:optional?',
  '/collect/:medium?',
  '/collections',
  '/collection/:slug',
  '/user/conversations',
  '/user/conversations/:conversationID',
  '/feature/:slug',
  '/identity-verification/processing',
  '/identity-verification/:id',
  '/order(2|s)/:orderID/respond',
  '/order(2|s)/:orderID/offer',
  '/order(2|s)/:orderID/shipping',
  '/order(2|s)/:orderID/payment',
  '/order(2|s)/:orderID/payment/new',
  '/order(2|s)/:orderID/review/counter',
  '/order(2|s)/:orderID/review',
  '/order(2|s)/:orderID/review/accept',
  '/order(2|s)/:orderID/review/decline',
  '/order(2|s)/:orderID/status',
  '/order(2|s)/:orderID',
  '/user/purchases',
  '/search/artists',
  '/search/collections',
  '/search/shows',
  '/search/fairs',
  '/search/institutions',
  '/search/galleries',
  '/search/categories',
  '/search/articles',
  '/search/auctions',
  '/search/more',
  '/search',
  '/viewing-room/:slug//works',
  '/viewing-room/:slug'
]
```

Reply:
Cool 👍 

This solves a dual issue here: https://github.com/artsy/force/blob/master/src/desktop/analytics/main_layout.js#L16-L33